### DISCUSSION TITLE
Scanned pdf auto alignment

### DISCUSSION THREAD

**Discussion by Zeyadas** [koreader/koreader#8286] [Ideas]
I just remembered using camscanner years ago, it's an app to scan documents using the phone camera then make a pdf file of it, the best features in the app was the auto crop it works exactly as the koreader page crop and the second feature was the auto alignment, it surprisingly gives an almost perfect aligned scanned page and here came my idea..
Most of the time I read pdfs (scanned) sometimes it needs alignment, so adding the auto alignment document page to KOReader will be helpful.

Didn't post it on feature request, I thought it would be better to post it here. 

![IMG_20210930_220546_505](https://user-images.githubusercontent.com/70109540/135516754-1bd8942a-876d-45bf-902e-68bd3a41031f.jpg)

**Comment by Frenzie** [comment]
There's dewarping in k2pdfopt, but on by default would probably be a bad idea and exposing it might litter the menus. But this looks like a very nice document all in all?

**Comment by poire-z** [comment]
It would be a nice feature indeed, and if k2pdfopt already supports it, it might be just a matter of playing and understanding how to enable it.
We have a bit of room in the PDF bottom menus, it could find its place in the 2nd menu.

(Pinging @Galunid , I remember you became familiar with kopt :) https://github.com/koreader/koreader-base/pull/1189 - in case you're in need of fun :)

**Reply by Frenzie** [reply]
Unlike what @Galunid did, enabling it should be quite trivial. ;-)

**Comment by yparitcher** [comment]
try changing `show = true`
https://github.com/koreader/koreader/blob/2b87b1d8ed0b71e8830615106b6d8fba86e67182/frontend/ui/data/koptoptions.lua#L570-L579

however see:
https://github.com/koreader/libk2pdfopt/blob/24b7e6bc136667c98feaa1b519a99dd880b05ebe/k2pdfoptlib/k2usage.c#L262-L266

> Note: de-warping, like auto-straighten and auto-crop, is intended for entire pages. It does not work on cropped areas. If you want it to work on cropped areas, you should run k2pdfopt in two passes--first to create selected crop areas (e.g. -mode crop), then to apply dewarping.\n"

**Reply by Frenzie** [reply]
Removing a line is significantly more trivial still. :rofl:
